### PR TITLE
Improve static analysis support

### PR DESCRIPTION
**Pull request by akva2**
In particular, this adds support for targets added using opm_add_test

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
i will self-merge this in soon as it is holding up other work.

File: cmake/Modules/OpmSatellites.cmake [hunk 1]
```diff
@@ -159,7 +159,11 @@ macro (opm_data satellite target dirname)
   set (${satellite}_INPUT_FILES)
   if (NOT PROJECT_SOURCE_DIR STREQUAL PROJECT_BINARY_DIR)
 	foreach (input_datafile IN LISTS ${satellite}_DATA)
-	  file (RELATIVE_PATH rel_datafile "${PROJECT_SOURCE_DIR}" ${input_datafile})
+	  if (IS_ABSOLUTE ${input_datafile})
+      file (RELATIVE_PATH rel_datafile "${PROJECT_SOURCE_DIR}" ${input_datafile})
+    else()
+      set(rel_datafile ${input_datafile})
+    endif()
```
Comment (bska) [inline cmake/Modules/OpmSatellites.cmake]: Is there some tab-vs-space indenting here?  Just casually reading this section it's not immediately clear that the conditional is within the `foreach` loop.

Comment (akva2) [reply cmake/Modules/OpmSatellites.cmake]: yes, it's a mess. i plan to do a reformatting pr later.

Comment (bska) [reply cmake/Modules/OpmSatellites.cmake]: >> Is there some tab-vs-space indenting here?
>
> yes, it's a mess. i plan to do a reformatting pr later.

Very well.  Merge at will.

File: cmake/Modules/UseStaticAnalysis.cmake [hunk 2]
```diff
@@ -34,16 +34,29 @@ function(add_static_analysis_tests sources includes)
...
+          add_test(NAME cppcheck+${name}
+                   COMMAND bin/cppcheck-test.sh ${CPPCHECK_PROGRAM} ${src} ${IPATHS}
+                   CONFIGURATIONS analyze cppcheck)
+        endif()
       endif()
       if(CLANGCHECK_FOUND AND CMAKE_EXPORT_COMPILE_COMMANDS)
-        add_test(NAME clang-check+${name}
-                 COMMAND bin/clang-check-test.sh ${CLANGCHECK_PROGRAM} ${src}
-                 CONFIGURATIONS analyze clang-check)
+        if(NOT TEST clang-check+${name})
```
Comment (bska) [inline cmake/Modules/UseStaticAnalysis.cmake]: I didn't pick up on this before, but `if (TEST` is not supported before CMake version 3.4.  This PR therefore effectively raises the minimum CMake requirement to version 3.4.

**Pull request merged**